### DISCUSSION TITLE
No re-render when re-validating and mutating happens at the same time

### DISCUSSION THREAD

**Discussion by Tom-Yem** [vercel/swr#1084] [Q&A]
I'm not sure if this is a bug, but here is what is happening.

I have a list of items in a table that each has an action button for updating. During an update action, i do an API update request and call mutate for updating the cache. Here is the UI:

![image](https://user-images.githubusercontent.com/53275974/112986309-9f930600-9161-11eb-89f9-e89557afe55a.png)

![image](https://user-images.githubusercontent.com/53275974/112986552-ed0f7300-9161-11eb-9067-f025d8c7cb0d.png)

This **always** works fine but it doesn't when the list is revalidating. For example:

After the page regains focus and immediately one action button is clicked, 
- the API request updates the data successfully. **(check)**
- _mutate_ function returns the updated cache. **(check)**
- But the **UI** doesn't re-render to show the new changes. **(fail)**

After the  page regains focus but instead of immediately, you wait for re-validating to finish _(isValidating resolve to false)_  and then do the action, then successfully the UI re-renders and shows the new changes.

Still not sure if this is how SWR is supposed to work but my current solution is to disable the action buttons while revalidating is true. But this could be bad experience for some one with slow connection.

My question is:
 ### why is it behaving like this?

**Comment by promer94** [comment]
I feel like this problem is similar to #1070 and #1074 which are fixed in #1075.

**Reply by Tom-Yem** [reply]
ok, it could be. I installed the latest version _yesterday_ (26-28 hours ago). I can see your commit for #1075 got merged to master 2 days ago.

![image](https://user-images.githubusercontent.com/53275974/113011248-e4c33200-9179-11eb-85aa-d2504b4ac761.png)

And i  see the releases covering up to #1055 
![image](https://user-images.githubusercontent.com/53275974/113011832-792d9480-917a-11eb-9fdd-2986975c69fe.png)

Could it be that your fix is not released yet?
